pkg/controllers/ocmcontroller/ocmagent.go

```diff
@@ -52,13 +60,32 @@ func InstallSelfManagementCluster(options options.ServerRunOptions) func(<-chan
 			}
 		}
 
-		go EnableSelfManagement(ctx, hubRestConfig, options.ControlplaneDataDir, clusterName)
+		kubeClient, err := kubernetes.NewForConfig(inClusterConfig)
+		if err != nil {
+			return err
+		}
+		apiserverURL, err := helpers.GetAPIServer(kubeClient)
```

Comment:
Maybe copy the methods from clusteradm repo. I feel it is a little heavy to depend on clusteradm repo only for util. What do you think?

Reply:
I can copy it. Will it make sense to move this common methods to the sdk-go repo? If so, what package path will make sense? If not, I can copy the code here.

Reply:
yes, make sense to move to sdk-go

Reply:
PTAL https://github.com/open-cluster-management-io/sdk-go/pull/23